### PR TITLE
Collected small changes and fixes

### LINKED ISSUE #3690: [BUG] fix mscg example segfaults
**Summary**

The MSCG example input deck in `examples/mscg` produces a segmentation fault in the force matching part of the run.

**LAMMPS Version and Platform**

8 Feb 2023

**Expected Behavior**

Run completes

**Actual Behavior**

Segmentation fault in second part of the run.

**Steps to Reproduce**

Compile LAMMPS with MSCG package included, run example in serial.

**Further Information, Files, and Links**

Example log was updated last in 2017. There is a difference in the force matching matrix dimensions. The range finder now finds 15 columns, the reference result 16.

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

This pull request combines multiple small changes and fixes.

**Related Issue(s)**

Fixes bugs reported in https://matsci.org/t/ndx2group-does-not-create-a-group/47545
Fixes #3690 

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

N/A

**Implementation Notes**

The following individual changes are included:
- update Plumed2 support with the recently released 2.8.2 and 2.7.6 versions
- enable building GPU package library with OpenMP enabled, if `-DBUILD_OMP=on` is used. Mention OpenMP support in docs
- provide access to `Force::special_lj` and `Force::special_coul` through the library interface and python module
- fix two bugs in `ndx2group` command
- download more recent snapshot for MSCG package that fixes an internal bug. 
- avoid race condition in "make install-python" with cmake.
- print warning when fix efield (and not fix efield/tip4p) is used with a TIP4P pair style
- must have a pair style defined for fix efield/tip4p to avoid segfault

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system

### CODE REVIEW

File: cmake/Modules/Packages/GPU.cmake [hunk 1]
```diff
@@ -407,6 +417,10 @@ elseif(GPU_API STREQUAL "HIP")
     target_compile_definitions(gpu PRIVATE -DMPI_GERYON -DUCL_NO_EXIT)
   endif()
   target_link_libraries(gpu PRIVATE hip::host)
+  if(BUILD_OMP)
```
Comment (rbberger) [inline cmake/Modules/Packages/GPU.cmake]: Wouldn't it be enough to add this to the end of the file once?

Comment (akohlmey) [reply cmake/Modules/Packages/GPU.cmake]: I have thought about that, but didn't like the disconnect. It is more obvious, if this command is where the other targets are added.

Comment (rbberger) [reply cmake/Modules/Packages/GPU.cmake]: So instead we duplicate code that we now have to maintain in multiple places?

Comment (akohlmey) [reply cmake/Modules/Packages/GPU.cmake]: Most of this file is an exercise in duplication...

Comment (akohlmey) [reply cmake/Modules/Packages/GPU.cmake]: "triplication" to be more accurate. The way I see it is the following: I already *know* that I have to check everything 3 times (CUDA, OpenCL, HIP), by moving the statement outside the sections, I have to remember to look at a fourth place. From that perspective, the current choice made more sense to me.

Comment (rbberger) [reply cmake/Modules/Packages/GPU.cmake]: I don't see that as a reason to add more cruft.

**Review by rbberger — DISMISSED** [review]
Please remove the redundancies. There is actually an opportunity here to further remove existing duplicated code.

**Comment by akohlmey** [comment]
> Please remove the redundancies. There is actually an opportunity here to further remove existing duplicated code.

done

**Review by sjplimp — APPROVED** [review]

**Review by rbberger — APPROVED** [review]

**Pull request merged**